src/main/java/roomescape/config/AdminOnlyInterceptor.java

```diff
@@ -0,0 +1,44 @@
+package roomescape.config;
+
+import jakarta.servlet.http.Cookie;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import org.springframework.web.servlet.HandlerInterceptor;
+import roomescape.member.domain.Member;
+import roomescape.member.service.MemberService;
+
+public class AdminOnlyInterceptor implements HandlerInterceptor {
+
+    private final MemberService memberService;
+
+    public AdminOnlyInterceptor(MemberService memberService) {
+        this.memberService = memberService;
+    }
+
+    @Override
+    public boolean preHandle(HttpServletRequest request, HttpServletResponse response, Object handler)
+            throws IOException {
+        String uri = request.getRequestURI();
+        if (!uri.startsWith("/admin")) {
+            return true;
+        }
+
+        Cookie[] cookies = request.getCookies();
+        if (cookies == null) {
+            response.sendRedirect("/login");
+            return false;
+        }
+        for (Cookie cookie : cookies) {
+            if ("token".equals(cookie.getName())) {
+                String token = cookie.getValue();
+                Member member = memberService.findByToken(token);
+                if (member.getRole().getId() == 2L) {
```

Comment:
Role에 id 개념이 왜 필요한건가요~?
아래가 더 깔끔하지 않을까요?

```suggestion
                if (member.isAdmin()) {
```

Reply:
네 member에게 관리자인지 아닌지 판단하게 하는 게 훨씬 가독성이 좋고, 응집도가 높은 코드인 것 같습니다.
그렇게 수정하였습니다.